### PR TITLE
ECO-3042: Add prefix to component

### PR DESCRIPTION
**Pull request by nikita-lysenko**
- Developer(s): @nikita-lysenko 

- Ticket: https://spryker.atlassian.net/browse/ECO-2954

#### Please confirm

- [ ] No new OS components - or they have been approved by the legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Adyen                 | minor                 |                       |

#### Release Notes

Added new frontend components `adyen-credit-card` with payment service provider prefix and deprecated the old one

-----------------------------------------

##### Change log

Improvements

* Introduced `adyen-credit-card` frontend component
* Deprecated `credit-card` frontend component
* Allow new OMS and Sales majors.

### CODE REVIEW

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    placeholder: {
+        color: '#d8d8d8'
+    },
+    validated: {
+        color: '#4fc2a0'
+    }
+};
+
+export default class AdyenCreditCard extends Component {
+    scriptLoader: ScriptLoader
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts]: Make all properties and getter `protected`.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% extends model('component') %}
+
+{% define config = {
+    name: 'adyen-credit-card',
+    tag: 'adyen-credit-card'
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig]: Add comma at the end. Check it everywhere.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.scss [hunk 3]
```diff
@@ -12,4 +12,4 @@
     }
 }
 
-@include adyen-credit-card();
+@include adyen-credit-card();
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.scss]: Add line break.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts [hunk 4]
```diff
@@ -71,4 +71,4 @@ export default class CreditCard extends Component {
     get configKey(): string {
         return this.getAttribute('client-config-key');
     }
-}
+}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: Add line break.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig [hunk 5]
```diff
@@ -39,4 +40,4 @@
             </div>
         </div>
     </div>
-{% endblock %}
+{% endblock %}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig]: Add line break.

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts (hunk 1 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts]: fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig (hunk 5 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.twig]: fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts (hunk 4 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.ts]: fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.scss (hunk 3 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/credit-card/credit-card.scss]: fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig (hunk 2 above)
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.twig]: fixed

File: src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    #{$name} {
+        &__input {
+            height: 35px;
+
+            iframe {
+                margin: 0;
+            }
+        }
+
+    @content;
```
[the remark this answers is not in the document]
Comment (nikita-lysenko) [reply src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.scss]: fixed

**Pull request merged**